modules/info.py

```diff
@@ -0,0 +1,109 @@
+    """
+
+    def __init__(
+        self,
+        *,
+        author: Union[discord.Member, discord.User],
+        entity: Union[discord.Member, discord.User, discord.Role, discord.TextChannel, discord.Guild]
+        # author: Union[discord.Member, discord.User],
+        # entity: Union[discord.Member, discord.User, discord.Role, discord.TextChannel, discord.Guild],
+    ):
+        super().__init__()
+        created_at: str = f"{discord.utils.format_dt(entity.created_at)} ({discord.utils.format_dt(entity.created_at, 'R')})"
+        if isinstance(entity, discord.Member) and entity.joined_at:
+            joined_at = f"\n\nJoined At: {discord.utils.format_dt(entity.joined_at)} ({discord.utils.format_dt(entity.joined_at, 'R')})"
+        else:
+            joined_at = ""
+
+        start: str = f"Name: {entity.name}\n\nID: {entity.id}\n\nCreated At: {created_at}{joined_at}"
+        description = start
+        if isinstance(entity, discord.Member):
+            description = start
+            self.set_thumbnail(url=entity.guild_avatar or entity.display_avatar or None)
+        elif isinstance(entity, discord.User):
+            description = start
+            self.set_thumbnail(url=entity.display_avatar or None)
+        elif isinstance(entity, discord.Role):
+            description += f"\n\nHoisted: {entity.hoist}\n\nMentionable: {entity.mentionable}\n\n"
+        elif isinstance(entity, discord.TextChannel):
+            description += f"\n\nCategory: {entity.category}\n\nNSFW: {entity.nsfw}"
+        else:  # Change to elif when other types are added
+            description += f"\n\nOwner: {entity.owner}"
+            self.set_thumbnail(url=entity.icon or None)
+
+        self.description = description
+        self.set_author(name=author.name, icon_url=author.display_avatar)
+        self.color = 0x7289DA
+
+
+class Information(core.Cog):
+    """The Mystbin file modification commands. Allows users to upload files to mystbin."""
+
+    def __init__(self, bot: core.Bot):
+        self.bot = bot
+
+    @commands.group(
+        name="information",
+        brief="Get information on a user, role, or channel",
+        aliases=["i", "info"],
+        invoke_without_command=True,
+    )
+    async def info(
+        self, ctx: core.Context, entity: Union[discord.Member, discord.User, discord.Role, discord.TextChannel, None] = None
+    ):
+        """Get information about a object
+        Args:
+            entity: The user, role, or TextChannel to get information about"""
+        embed = InformationEmbed(author=ctx.author, entity=entity or ctx.author)
+        await ctx.send(embed=embed)
+
+    @info.command(name="guild", brief="Get the current guild's information.")
+    @commands.guild_only()
+    async def guild_info(self, ctx: core.Context):
+        if ctx.guild is None:
+            raise commands.CheckFailure("This command must be ran in a guild.")  # Make pyright stop throwing fits
```

Comment:
Annotate `ctx` as `core.GuildContext` since we are guarded correctly with the `guild_only()` decorated check. This removes the need for lines 102-103.

```suggestion
    @commands.guild_only()
    async def guild_info(self, ctx: core.GuildContext) -> None:
```